github-bugzilla-content.js

```diff
@@ -8,7 +8,7 @@
  */
 
 // Regexp to match against PR title
-const BUG_RE = /\b(ticket|bug|tracker item|issue)s?:? *([\d ,\+&#and]+)\b/i;
+const BUG_RE = /\b(ticket|bug|tracker item|issue)-?s?:? *([\d ,\+&#and]+)\b/i;
```

Comment:
I don't think this change works. The `s?` allows for plurals (e.g. "bug" and "bugs") and the `:?` allows for a colon (e.g. "bug 111" and "bug: 1111"). It also allows for a list of bugs. So all of these should match:

* "bug 111"
* "bug: 111"
* "bugs 111"
* "bugs 111, 222, & 333"
* "bugs 111, 222, and 333"
* "bugs: 111, 222"

We want to add support for this:

* "bug-111"

We probably want to think about multiple bugs, too. How would that structure work with multiple bugs? Maybe something like this?:

* "bug-111, bug-222, bug-333"

That's a really different structure. I think we'd need to do something like this:

```suggestion
const BUG_RE = /\b(?:(ticket|bug|tracker item|issue)s?:? *([\d ,\+&#and]+)|(ticket|bug|issue)s?-(\d+))\b/i;
```

I think that allows for the different structures without changing the match group indexes. rob-bugson has no tests, so it's kind of annoying to verify that. :cry:

Reply:
> I don't think this change works.

I tested for the singular case of `bug-*`[1] initially, and I just tested for the plural case, `bug-*, bug-*, bug-*`, and those work for me [2]. How are you testing them?

[1]:
```sh
bdanforth ~/Projects/rob-bugson (main) $ node
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> var re = /\b(ticket|bug|tracker item|issue)s?:? *([\d ,\+&#and]+)\b/i;
undefined
> var test = "bug 12345";
undefined
> re.test(test)
true
> var testHyphen = "bug-12345";
undefined
> re.test(testHyphen);
false
> var reModified = /\b(ticket|bug|tracker item|issue)-?s?:? *([\d ,\+&#and]+)\b/i;
undefined
> reModified.test(testHyphen);
true
> 
```

[2]:
```
bdanforth ~/Projects/rob-bugson (support-hyphens-in-bug-re) $ node
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> var reModified = /\b(ticket|bug|tracker item|issue)-?s?:? *([\d ,\+&#and]+)\b/i
undefined
> var testHyphen = "bug-12345";
undefined
> reModified.test(testHyphen);
true
> var testHyphenMultiple = "bug-12345,bug-23456";
undefined
> reModified.test(testHyphenMultiple);
true
> var testHyphenMultiple = "bug-12345, bug-23456";
undefined
> reModified.test(testHyphenMultiple);
```

Reply:
What does `true` mean? Does it mean that the regex matches the entire string? Do the match groups work?

Reply:
I'm eye-balling the regex--not testing it. But it looks like the change you made also matches these things:

* "bug-s 111"
* "bug-: 111"
* "bug-s: 111"

We don't really want to match that. Is it a big deal? Probably not, but as the regex evolves over time, it's easier to maintain if it matches only the intended strings.

I don't see how it could match something like "bug-111, bug-222" since the entire pattern doesn't repeat. So it's puzzling that passes in your test.

Reply:
>  What does true mean? Does it mean that the regex matches the entire string? Do the match groups work?

`true` means it found a match in the string for the regexp. Admittedly, this is not all we need to know. We need the list of matches, which comes from `exec`, which is what is used in the extension, and what I should be using to test.

I've since rebased this PR off of #54 and added some tests to cover the test cases you indicated here.

> We probably want to think about multiple bugs, too. How would that structure work with multiple bugs? Maybe something like this?:
>    "bug-111, bug-222, bug-333"

The equivalent of this scenario for spaces is also not currently supported by the existing regexp and function (i.e. "bug 111, bug 222, bug 333"). I've filed #58 to cover this scenario, as I think this will require changes to both the regexp and the function that uses it, and I couldn't get something to work in my one hour timebox. Would it be okay with you to descope this part? Other than that I think this PR is ready.

Reply:
This conversation is continued in this comment: https://github.com/willkg/rob-bugson/pull/51#pullrequestreview-1730964163 and the subsequent two commits to the PR. Marking it as resolved. Thanks WillKG!